cc_utilities/redcap_sync.py

```diff
@@ -260,21 +264,80 @@ def handle_cdms_matching(
     logger.info(f"Checking CDMS for DOB and ID matches on {len(df.index)} records.")
     df = df.dropna(subset=[DOB_FIELD])
     matching_ids = get_matching_cdms_patients(df, db_url, external_id_col)
-    matched_records, unmatched_records = select_records_by_cdms_matches(
+    accept_records, reject_records = split_records_by_accepted_external_ids(
         df, redcap_records, matching_ids, external_id_col
     )
-    if len(unmatched_records) > 0:
-        unmatched_records = add_integration_status_columns(
-            unmatched_records,
+    if len(reject_records) > 0:
+        reject_records = add_integration_status_columns(
+            reject_records,
             status=REDCAP_REJECTED_PERSON,
             reason=f"mismatched {DOB_FIELD} and {external_id_col}",
         )
         import_records_to_redcap(
-            unmatched_records,
+            reject_records,
             redcap_api_url=redcap_api_url,
             redcap_api_key=redcap_api_key,
         )
-    return matched_records
+    return accept_records
+
+
+def get_commcare_cases_with_acceptable_interview_dispositions(
+    df, external_id_col, cc_api_key, cc_user_name, project_slug
+):
+    """
+    Look up existing cases in CommCare and compare with accepted interview_disposition
+    values. This should allow us to filter out cases which have already been filled
+    out by a case investigator so that patient surveys do not override case
+    investigator's data. We will reject the cases not matching and send
+    them back to REDCap, and otherwise continue to send them to CommCare.
+    """
+    accepted_external_ids = []
+    external_ids = df[external_id_col].to_list()
+    for ext_id in external_ids:
+        cases = get_commcare_cases_by_external_id_with_backoff(
```

Comment:
A comment about why we query CommCare directly (instead of looking at the SQL DB mirror) could be helpful. E.g., if it turns out this is too slow and we need to query the SQL DB instead, understanding the tradeoff there will be good (the SQL DB mirror is up to 5 minutes behind CommCare itself, so we might not see if a case investigator has started on a case).

Reply:
I think the answer is that the interview disposition was not expected to be in the SQL Mirror at all - Sherjeel told me I would just find it in CommCare (see this slack thread: https://caktusgroup.slack.com/archives/C016RMY7YAC/p1622136245003500)